spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/TcpOutboundGateway.java

```diff
@@ -155,6 +156,9 @@ protected Object handleRequestMessage(Message<?> requestMessage) {
 					logger.debug("released semaphore");
 				}
 			}
+			if (this.connectionFactory instanceof CloseDeferrable) {
+				((CloseDeferrable) this.connectionFactory).closeDeferred(connectionId);
```

Comment:
Yes; the responsibility is wrong; but fixing it would be too big a change for a point release (hence the reference to INT-3654).

Thanks